zetaclient/types/event.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import (
+	"github.com/zeta-chain/zetacore/pkg/coin"
+)
+
+// InboundEvent represents an inbound event
+// TODO: we should consider using this generic struct when it applies (e.g. for Bitcoin, Solana, etc.)
```

Comment:
Need issue link for TODO

Reply:
https://github.com/zeta-chain/node/pull/2465/commits/97fd39920f82372ac94065c92c07d9633d93eeba